### DISCUSSION TITLE
renogy commands

### DISCUSSION THREAD

**Discussion by riker65** [wildmountainfarms/solarthing#218] [General]
Hi
I am wondering if there is a way to reset / restart the charger via command? 
sometimes the rover got stuck and I have to reset it via disconnecting.

Would be much easier if this would be possible via command  with nc....

Thanks

**Comment by retrodaredevil** [comment]
If you have the Rover config server setup (https://solarthing.readthedocs.io/en/latest/config/config-server.html), then you should be able to give it a `factoryreset !!` command. Like so:

```
factoryreset !!
```

The `!!` is a safeguard I put in place. Feel free to run something like `factoryreset lololol`, and it should give you an error. Please note that the number of arguments is important (it won't recognize `factoryreset arg1 arg2` or just `factoryreset`).

Please keep in mind that a factory reset will **RESET ALL YOUR SETTINGS** and reset any historical data that the rover stores in its internal memory (unrelated to historical data SolarThing itself stores). I also haven't tested the factory reset in a long time. I think I personally tested it once, and I assume that it still works as intended.

Now, what I don't understand is why that would be necessary. What do you mean that the rover "got stuck"? I've never had any problem like that.
